src/interface.jl

```diff
@@ -428,9 +428,14 @@ ManifoldsBase.exp(G::AbstractLieGroup, ::Any, ::Any)
 
 @doc "$_doc_exp"
 function ManifoldsBase.exp!(G::AbstractLieGroup, h, g, X)
-    exp!(G, h, X)
-    compose!(G, h, g, h)
-    return h
+    if Base.mightalias(g, h)
```

Comment:
I was not aware of this function, that is really neat!

While it might resolve the issue here, I think this would even be nicer to not add it here but in every `compose!` implementation.

Reply:
@mateuszbaran suggested the function previously. 

`compose!` might also have aliasing problems, however the `exp!` here overwrites the value of `h` and therefore `g`.

Here is an example to further clarify:
```julia
julia> M = TranslationGroup(2);

julia> g = [2., 3];

julia> h = g;

julia> exp!(M, h, X)
2-element Vector{Float64}:
 0.1
 0.0
```
`g` is now set to the wrong value of `[0.1, 0.0]`